### PR TITLE
Add support for viewing zstd compressed files

### PR DESCRIPTION
**Pull request by RonCruz**
The following changes are for a new feature request to support viewing (zstd)[https://facebook.github.io/zstd/] compressed files. I took an initial stab at the implementation for this. I'm new to Groovy but it seemed simple enough following the implementation for existing file types.

There is one caveat: accepting this PR does add a new dependency to the project (https://github.com/luben/zstd-jni).

### CODE REVIEW

**Comment by Camork** [comment]
Hi @RonCruz , can we use the pure java implementation [aircompressor](https://mvnrepository.com/artifact/io.airlift/aircompressor) ? because the JNI implementation will bring every platform's native libraries , that cause the packed plugin size to be huge.

**Comment by RonCruz** [comment]
@Camork Sure. I've pushed a change to migrate away from ztsd-jni to your suggested library.

File: src/main/groovy/com/github/camork/filesystem/zstd/ZstdFile.groovy [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+    ZstdFile(File file) {
+        this.file = file
+    }
+
+    @Override
+    Map<String, ?> createEntriesInfoMap() {
+        Map entries = new HashMap<String, ?>()
+        EntryInfo root = ArchiveUtils.createRootEntry()
+        entries.put('', root)
+        entries.put('contents', new EntryInfo('contents', false, file.length(), file.lastModified(), root))
```
Comment (Camork) [inline src/main/groovy/com/github/camork/filesystem/zstd/ZstdFile.groovy:21]: Sorry for the later, It seem that the implement isn't completed, I will take over the left.

**Comment by Camork** [comment]
Merged to master, I had mixed up archiver and compressor before, I need to refactor the code so that it can auto-detect the compressor and archiver by it's metadata, not only the file name.

**Pull request closed without merging**